fix(docs): replace twitter logo/links with x logo/links

dgz9 opened the pull request:
Closes # N/A

## 📝 Description

> This PR updates all Twitter references to X (formerly Twitter) in the NextUI documentation. This includes changing the Twitter logo to the X logo and updating all twitter.com links to x.com.

## ⛳️ Current behavior (updates)

> Currently, the NextUI documentation uses the old Twitter bird logo and links to twitter.com for social media references.

## 🚀 New behavior

> - The Twitter bird logo has been replaced with the new X logo.
> - All twitter.com links have been updated to x.com.
> - The TwitterIcon component has been renamed to XIcon and its SVG path has been updated to represent the X logo.

## 💣 Is this a breaking change (Yes/No):

> No

## 📝 Additional Information

> N/A

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `XIcon` component for the updated branding.
  
- **Bug Fixes**
	- Updated various URLs from Twitter to X to reflect the platform rebranding.

- **Documentation**
	- Revised links in README and other documentation to point to the new X domain. 

- **Style**
	- Adjusted the visual presentation of the `LinkIcon` in the `FeaturesGrid` component by removing the specific class name.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

wingkwong reviewed (changes requested):
`TwitterIcon` can be removed since it's not in use.

dgz9 commented:
> `TwitterIcon` can be removed since it's not in use.

Will do tonight.

Do you want me to remove all references of Twitter and replace with X or just keep it at the logo and links?

wingkwong commented:
I mean just `TwitterIcon` in  `apps/docs/components/icons/social.tsx` since you replaced it with `XIcon`. `TwitterIcon` is not in use anymore.

dgz9 commented:
I was wondering about all the references to twitter, but I left it for now.

I removed the `TwitterIcon` and changed a few more name references (even though it looks weird as just X).

wingkwong commented:
Is it ready for review or you still have further changes?

dgz9 commented:
Ready for review. Let me know if you want any other changes.

wingkwong commented on apps/docs/components/navbar.tsx:
```diff
@@ -325,12 +325,12 @@ export const Navbar: FC<NavbarProps> = ({children, routes, mobileRoutes = [], sl
             className="p-1"
             href={siteConfig.links.twitter}
-            onPress={() => handlePressNavbarItem("Twitter", siteConfig.links.twitter)}
+            onPress={() => handlePressNavbarItem("X", siteConfig.links.twitter)}
```
This change would change the trace event title in analytics. please rollback.

wingkwong commented on .changeset/slimy-cats-brush.md:
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/user": patch
+"@nextui-org/react": patch
```
since the changes don't involve version bump, you may remove these two lines.

wingkwong reviewed (changes requested):
final round of changes

wingkwong reviewed (changes requested):
X icon in light theme is not visible.

![image](https://github.com/user-attachments/assets/affff415-2bc0-4ed3-9c54-68e59a535239)
![image](https://github.com/user-attachments/assets/6ea7c9a0-85d5-4ab5-a794-f6d5c0da6300)

dgz9 commented:
Fixed light theme icon. Also noticed the feature grid icon was not working for light more, changed that too.
